argo/base_v3/kustomization.yaml

```diff
@@ -0,0 +1,102 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../base/cluster-role-binding.yaml
+- ../base/cluster-role.yaml
+- ../base/config-map.yaml
+- ../base/crd.yaml
+- ../base/deployment.yaml
+- ../base/service-account.yaml
+- ../base/service.yaml
+- ../overlays/istio/virtual-service.yaml
+- ../overlays/application/application.yaml
+commonLabels:
+  kustomize.component: argo
+  app.kubernetes.io/component: argo
+  app.kubernetes.io/name: argo
+  app.kubernetes.io/part-of: kubeflow
+images:
+- name: argoproj/argoui
+  newName: argoproj/argoui
+  newTag: v2.3.0
+- name: argoproj/workflow-controller
+  newName: argoproj/workflow-controller
+  newTag: v2.3.0
+configMapGenerator:
+- name: workflow-controller-parameters
+  env: ../base/params.env
+generatorOptions:
+  disableNameSuffixHash: true
```

Comment:
Probably something we didn't consider, adding the option here disables name suffix generation globally: as shown in test failure.

Reply:
```
        kustomize.component: jupyter-web-app                                               kustomize.component: jupyter-web-app
    template:                                                                          template:
      metadata:                                                                          metadata:
        annotations:                                                                       annotations:
          sidecar.istio.io/inject: "false"                                                   sidecar.istio.io/inject: "false"
        labels:                                                                            labels:
          app: jupyter-web-app                                                               app: jupyter-web-app
          kustomize.component: jupyter-web-app                                               kustomize.component: jupyter-web-app
      spec:                                                                              spec:
        containers:                                                                        containers:
        - env:                                                                             - env:
          - name: USERID_HEADER                                                              - name: USERID_HEADER
            valueFrom:                                                                         valueFrom:
              configMapKeyRef:                                                                   configMapKeyRef:
                key: userid-header                                                                 key: userid-header
X                name: kubeflow-config                                                              name: kubeflow-config-4bkkg42k5m
          - name: USERID_PREFIX                                                              - name: USERID_PREFIX
            valueFrom:                                                                         valueFrom:
              configMapKeyRef:                                                                   configMapKeyRef:
                key: userid-prefix                                                                 key: userid-prefix
X                name: kubeflow-config                                                              name: kubeflow-config-4bkkg42k5m
          image: gcr.io/kubeflow-images-public/jupyter-web-app:vmaster-gd9be4b9e             image: gcr.io/kubeflow-images-public/jupyter-web-app:vmaster-gd9be4b9e
          name: jupyter-web-app                                                              name: jupyter-web-app
          ports:                                                                             ports:
          - containerPort: 5000                                                              - containerPort: 5000
          volumeMounts:                                                                      volumeMounts:
          - mountPath: /etc/config                                                           - mountPath: /etc/config
            name: config-volume                                                                name: config-volume
        serviceAccountName: jupyter-web-app-service-account                                serviceAccountName: jupyter-web-app-service-account
        volumes:                                                                           volumes:
        - configMap:                                                                       - configMap:
            name: jupyter-web-app-jupyter-web-app-config-5cc2fchd2g                            name: jupyter-web-app-jupyter-web-app-config-5cc2fchd2g
          name: config-volume                                                                name: config-volume
   TestKustomize: test_util.go:176: Expected not equal to actual
```

Reply:
Hmm, actually this seems like a bug in `tests/ make generate` script. I tried in master branch, it will also get this problem.

Reply:
Something like something else might be going on. Was there a bad merge for the jupyter manifests somewhere?

Reply:
@jlewi I'm not sure, I have limited knowledge in the genreate script. Can you help me look into this?

For this PR, I can manually revert the unexpected snapshot files.